src/Entities/Chunk.php

```diff
@@ -0,0 +1,386 @@
+<?php
+/**
+ * Chunk Entity
+ *
+ * @package juvo\AS_Processor
+ */
+
+namespace juvo\AS_Processor\Entities;
+
+use Exception;
+use juvo\AS_Processor\DB;
+
+/**
+ * The chunk entity class
+ */
+class Chunk
+{
+    use DB;
+
+    /**
+     * @var int
+     */
+    private int $chunk_id;
+
+    /**
+     * @var int|null
+     */
+    private ?int $action_id = null;
+
+    /**
+     * @var string|null
+     */
+    private ?string $name = null;
+
+    /**
+     * @var string|null
+     */
+    private ?string $group = null;
+
+    /**
+     * @var string|null
+     */
+    private ?string $status = null;
+
+    /**
+     * @var array<mixed>|null
+     */
+    private ?array $data = null;
+
+    /**
+     * @var string|null
+     */
+    private ?string $start = null;
+
+    /**
+     * @var string|null
+     */
+    private ?string $end = null;
+
+    /**
+     * @var string|null
+     */
+    private ?string $duration = null;
+
+    /**
+     * @var bool
+     */
+    private bool $is_data_fetched = false;
+
+    /**
+     * Constructor
+     *
+     * @param int|null $chunk_id The chunk ID.
+     */
+    public function __construct( ?int $chunk_id = null ) {
+        if ( empty( $chunk_id ) ) {
+            $this->chunk_id = $this->create_chunk();
+        } else {
+            $this->chunk_id = $chunk_id;
+        }
```

Comment:
Why is this needed? The logic should more be like "if id is set get chunk from id". To save in the db rename "create" to "save". It should be called explicitly. This is how WC does it for products